cookie.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @typedef {{
+ *      domain: string,
+ *      path: string,
+ *      secure: boolean,
+ *      ?expires: number|Date,
+ * }} mojave.CookieOptions
+ */
+
+
+/**
+ * Sets a cookie's value. If the value is an object or array, it'll be JSON.stringify and saved as a string
+ *
+ * @param {string} key
+ * @param {*} value
+ * @param {mojave.CookieOptions} options
+ */
+export function setCookie (key, value, options = {})
+{
+    options = Object.assign({
+        path: "/",
+        secure: window.location.protocol === "https",
+    }, options);
+
+    if (typeof options.expires === "number")
+    {
+        options.expires = new Date(new Date() * 1 + options.expires * 864e+5);
+    }
+
+    options.expires = typeof options.expires !== "undefined"
+        ? options.expires.toUTCString()
+        : "";
+
+    const encodedKey = encodeCookieKey(key);
+    const encodedValue = encodeCookieValue(value);
+    const encodedOptions = encodeCookieOptions(options);
+
+    document.cookie = `${encodedKey}=${encodedValue};${encodedOptions.join(" ;")}`;
+}
+
+
+/**
+ * Retrieves a cookie's value
+ *
+ * @param {string} key
+ *
+ * @returns {*|null}
+ */
+export function getCookie (key)
+{
+    const cookies = document.cookie.split("; ");
+
+    for (let i = 0; i < cookies.length; i++)
+    {
+        const parts = cookies[i].split("=");
+        const cookieName = decodeURIComponent(parts[0]);
+
+        let cookieValue = parts.slice(1).join("=");
+        if (cookieValue.charAt(0) === '"')
+        {
+            cookieValue = cookieValue.slice(1, -1);
+        }
+
+        const decodedCookieValue = decodeURIComponent(cookieValue);
+
+        if (cookieName === key)
+        {
+            try
+            {
+                return JSON.parse(decodedCookieValue);
+            }
+            catch (err)
+            {
+                // Silently swallow any JSON parse errors
+                return null;
+            }
+        }
+    }
+
+    return null;
+}
+
+
+/**
+ * Removes the given cookie
+ *
+ * @param {string} key
+ */
+export function removeCookie (key)
+{
+    setCookie(key, "", {
+        expires: -1,
+    });
+}
+
+
+
+/**
+ * @private
+ * @param {string} key
+ *
+ * @return {string}
+ */
+function encodeCookieKey (key)
+{
+    return encodeURIComponent(String(key))
```

Comment:
You shouldn't use the boxed types in JS. If you want to convert to string, then just use `"" + key`